### PR TITLE
Added new options to system tray

### PR DESCRIPTION
**Pull request by fayeed**
- Open explorer where the logs are instead of copying them to the desktop.
- Added a way to report issues by opening GitHub.
- Added a way to copy the executor URL.

### CODE REVIEW

File: src-tauri/src/config.rs [hunk 1]
```diff
@@ -26,4 +26,14 @@ pub fn app_url(port: u16) -> String {
 #[cfg(not(feature = "custom-protocol"))]
 pub fn app_url(port: u16) -> String {
     format!("http://localhost:3000?ad4m=ws://localhost:{}/graphql", port)
+}
+
+#[cfg(feature = "custom-protocol")]
+pub fn executor_url(port: u16) -> String {
```
Comment (kaichaosun) [inline src-tauri/src/config.rs]: This doesn't make sense, since only port is used by ad4min frontend

File: src-tauri/src/system_tray.rs [hunk 2]
```diff
@@ -1,22 +1,30 @@
...
-use directories::UserDirs;
+use crate::logs::open_logs_folder;
+use crate::util::open_url;
+use tauri::ClipboardManager;
 
 pub fn build_system_tray() -> SystemTray {
     let show_ad4min = CustomMenuItem::new("show_ad4min".to_string(), "Show Ad4min");
     let quit = CustomMenuItem::new("quit".to_string(), "Quit");
-    let copy_logs = CustomMenuItem::new("copy_logs".to_string(), "Copy Logs");
+    let open_logs = CustomMenuItem::new("open_logs".to_string(), "Open Logs");
```
Comment (kaichaosun) [inline src-tauri/src/system_tray.rs]: copy_url seems not useful as a system tray menu item.
open_logs and report_issue should be a menu item (help -> open logs, report issue), not a system tray item.

File: src-tauri/src/util.rs [hunk 3]
```diff
@@ -8,3 +8,11 @@ pub fn find_port(start_port: u16, end_port: u16) -> u16 {
   panic!("No open port found between: [{:?}, {:?}]", start_port, end_port);
 }
 
+#[tauri::command]
```
Comment (kaichaosun) [inline src-tauri/src/util.rs]: this seems not an command.

File: src-tauri/src/system_tray.rs [hunk 4]
```diff
@@ -47,15 +55,24 @@ pub fn handle_system_tray_event(app: &AppHandle<Wry>, event_id: String, port: u1
...
+            report_issue();
         }
         "quit" => {
             app.exit(0);
         }
         _ => log::error!("Event is not defined."),
     }
 }
+
+#[tauri::command]
```
Comment (kaichaosun) [inline src-tauri/src/system_tray.rs]: put the command in a file src/commands/help maybe?

File: src-tauri/src/util.rs (hunk 3 above)
Comment (kaichaosun) [inline src-tauri/src/util.rs]: This is not resolved right?

also you can inlice this method in report issue command.

Comment (fayeed) [reply src-tauri/src/util.rs]: I have kept it as a command so we can open an external URL from the app

File: src-tauri/src/main.rs [hunk 5]
```diff
@@ -72,8 +75,9 @@ fn main() {
     let builder_result = tauri::Builder::default()
         .manage(state)
         .menu(build_menu())
+        .on_menu_event(|event| handle_menu_event(event.menu_item_id(), event.window()))
         .system_tray(build_system_tray())
-        .invoke_handler(tauri::generate_handler![get_port])
+        .invoke_handler(tauri::generate_handler![get_port, open_logs_folder])
```
Comment (kaichaosun) [inline src-tauri/src/main.rs]: I think open_logs_folder should not be here.

File: src-tauri/src/commands/help.rs [hunk 6]
```diff
@@ -0,0 +1,16 @@
+use crate::config::data_path;
+
+#[tauri::command]
```
Comment (kaichaosun) [inline src-tauri/src/commands/help.rs]: tauri command is used for IPC request, for menu reaction, the doc is here https://tauri.studio/v1/guides/features/menu/

File: src-tauri/src/system_tray.rs [hunk 7]
```diff
@@ -1,24 +1,21 @@
-use crate::app_url;
+use crate::{app_url};
 
-use crate::config::log_path;
 use tauri::{
     AppHandle, CustomMenuItem, Manager, SystemTray, SystemTrayMenu, SystemTrayMenuItem,
     WindowBuilder, WindowUrl, Wry,
 };
+use tauri::ClipboardManager;
```
Comment (kaichaosun) [inline src-tauri/src/system_tray.rs]: Is this still needed?

File: src-tauri/src/commands/help.rs [hunk 8]
```diff
@@ -0,0 +1,14 @@
+use crate::config::data_path;
+
+pub fn report_issue() {
```
Comment (kaichaosun) [inline src-tauri/src/commands/help.rs]: The functions better to put in menu.rs.

File: src-tauri/Cargo.toml [hunk 9]
```diff
@@ -23,8 +23,10 @@ nix = "0.23.1"
 portpicker = "0.1.1"
 libc = "0.2.121"
 directories = "4.0.1"
+opener = "0.5.0"
+open = "2.0.1"
 sysinfo = "0.23.12"
-tauri = { version = "1.0.0-rc.7", features = ["api-all", "gtk-tray", "system-tray", "wry"], default-features = false }
+tauri = { version = "1.0.0-rc.7", features = ["api-all", "clipboard", "gtk-tray", "system-tray", "wry"], default-features = false }
```
Comment (kaichaosun) [inline src-tauri/Cargo.toml]: `clipboard` not used.

**Review by kaichaosun — APPROVED** [review]

**Pull request merged**